torchtnt/framework/callbacks/dcp_saver.py

```diff
@@ -193,13 +193,15 @@ def _checkpoint_impl(
             with get_timing_context(state, f"{self.__class__.__name__}.async_save"):
                 # Redundant check for safety
                 self._wait(log_warning=True)
-                self._prev_snapshot = dcp.async_save(
+                prev_snapshot = dcp.async_save(
                     state_dict={"app_state": MultiStateful(app_state)},
                     checkpoint_id=checkpoint_id,
                     process_group=self._process_group,
                     storage_writer=storage_writer,
                     planner=planner,
                 )
+                assert isinstance(prev_snapshot, Future)
```

Comment:
Could we get any errors because of this `isinstance` check?

If this is just for pyre, then might be better to use `cast` which will not affect runtime.